scripts/src/java/org/oppia/android/scripts/RegexPatternValidationCheck.kt

```diff
@@ -0,0 +1,263 @@
+ *  file naming patterns are not present in the codebase.
+ */
+class RegexPatternValidationCheck {
+  companion object {
+    @JvmStatic
+    fun main(vararg args: String) {
+      /** path of the repo to be analyzed. */
+      val repoPath = args[0] + "/"
+
+      /** a list of all allowed directories in the repo to be analyzed. */
+      val allowedDirectories = args.drop(1)
+
+      /** a list of all files in the repo to be analyzed. */
+      val searchFiles = RepoFile.collectSearchFiles(
+        repoPath = repoPath,
+        allowedDirectories = allowedDirectories
+      )
+
+      /** check if the repo has any filename failure. */
+      val hasFilenameCheckFailure = retrieveFilenameChecks()
+        .fold(initial = false) { isFailing, check ->
+          val checkFailed = checkProhibitedFileNamePattern(
+            repoPath = repoPath,
+            searchFiles = searchFiles,
+            prohibitedFilenameObj = check,
+          )
+          return@fold isFailing || checkFailed
+        }
+
+      /** check if the repo has any file content failure. */
+      val hasFileContentCheckFailure = retrieveFileContentChecks()
+        .fold(initial = false) { isFailing, check ->
+          val checkFailed = checkProhibitedContent(
+            repoPath = repoPath,
+            searchFiles = searchFiles,
+            prohibitedContentObj = check,
+          )
+          return@fold isFailing || checkFailed
+        }
+
+      if (hasFilenameCheckFailure || hasFileContentCheckFailure) {
+        throw Exception(ScriptResultConstants.REGEX_CHECKS_FAILED)
+      } else {
+        println(ScriptResultConstants.REGEX_CHECKS_PASSED)
+      }
+
+    }
+
+    /**
+     * Retrieves all filename checks.
+     *
+     * @return a list of all the FilenameChecks
+     */
+    private fun retrieveFilenameChecks(): List<FilenameCheck> {
+      return getProto(
+        textProtoFileName = "filename_pattern_validation_checks.pb",
+        proto = FilenameChecks.getDefaultInstance()
+      ).getFilenameChecksList()
+    }
+
+    /**
+     * Retrieves all file content checks.
+     *
+     * @return a list of all the FileContentChecks
+     */
+    private fun retrieveFileContentChecks(): List<FileContentCheck> {
+      return getProto(
+        textProtoFileName = "file_content_validation_checks.pb",
+        proto = FileContentChecks.getDefaultInstance()
+      ).getFileContentChecksList()
+    }
+
+    /**
+     * helper function to parse the textproto file to a proto class.
+     *
+     * @param textProtoFileName name of the textproto file to be parsed
+     * @param proto instance of the proto class
+     * @return proto class from the parsed textproto file
+     */
+    private fun <T : MessageLite> getProto(textProtoFileName: String, proto: T): T {
+      val protoBinaryFile = File("scripts/assets/$textProtoFileName")
+      val builder = proto.newBuilderForType()
+
+      @Suppress("UNCHECKED_CAST")
```

Comment:
can we remove this suppressing?
WDYT?

Reply:
Removing this, gives a warning during the compilation of the script. This is actually added to suppress that warning. We are doing the same thing in the existing code too.
https://github.com/oppia/oppia-android/blob/f9eb3cdcba174fdd78690af33ffe3c7a8b547a10/utility/src/main/java/org/oppia/android/util/extensions/BundleExtensions.kt#L38

Reply:
COuld you add what warning it is giving after removing it?

Reply:
```
scripts/src/java/org/oppia/android/scripts/RegexPatternValidationCheck.kt:106:17: warning: unchecked cast: MessageLite! to T
      }.build() as T
```